connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginDesc.java

```diff
@@ -110,11 +110,18 @@ public int hashCode() {
     }
 
     @Override
-    public int compareTo(PluginDesc<T> other) {
+    public int compareTo(PluginDesc<?> other) {
         int nameComp = name.compareTo(other.name);
         int versionComp = encodedVersion.compareTo(other.encodedVersion);
         // isolated plugins appear after classpath plugins when they have identical versions.
         int isolatedComp = Boolean.compare(other.loader instanceof PluginClassLoader, loader instanceof PluginClassLoader);
-        return nameComp != 0 ? nameComp : (versionComp != 0 ? versionComp : isolatedComp);
+        // choose an arbitrary order between different locations and types
+        int loaderComp = location.compareTo(other.location);
```

Comment:
This is causing an NPE during unit tests (see CI results):
```suggestion
        int loaderComp = Objects.compare(location, other.location, String::compareTo);
```

Reply:
We should probably make all of the comparisons here null-safe, or enforce that the relevant fields are non-null in the constructor.

Reply:
The one line change was appealing, but as all of these shouldn't be null in production, i'll force the tests to also provide non-null arguments and locations.